libs/jwst-codec/src/codec/integer.rs

```diff
@@ -0,0 +1,185 @@
+use super::*;
+use byteorder::WriteBytesExt;
+use nom::{number::complete::be_u8, Needed};
+use std::io::{Error, Write};
+
+pub fn read_var_u64(input: &[u8]) -> IResult<&[u8], u64> {
+    let mut shift = 0;
+    let mut result: u64 = 0;
+    let mut rest = input;
+    loop {
+        let (tail, byte) = match be_u8(rest) {
```

Comment:
`let (tail, byte) = be_u8(rest).map_err(|err| if let nom::Err::Incomplete(Needed::Size(n)) = err { nom::Err::Incomplete(Needed::Size(n.saturating_add(1))) } else { err })`

use `saturating_add` to avoid `unwrap` here